vehicles/src/modules/shopping-cart/shopping-cart.service.ts

```diff
@@ -0,0 +1,254 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { NotBrackets, Repository, SelectQueryBuilder } from 'typeorm';
+import { LogAsyncMethodExecution } from '../../common/decorator/log-async-method-execution.decorator';
+import { ApplicationStatus } from '../../common/enum/application-status.enum';
+import { Directory } from '../../common/enum/directory.enum';
+import {
+  ClientUserAuthGroup,
+  IDIRUserAuthGroup,
+  IDIR_USER_AUTH_GROUP_LIST,
+  UserAuthGroup,
+} from '../../common/enum/user-auth-group.enum';
+import { Permit as Application } from '../permit-application-payment/permit/entities/permit.entity';
+import { AddToShoppingCartDto } from './dto/request/add-to-shopping-cart.dto';
+import { UpdateShoppingCartDto } from './dto/request/update-shopping-cart.dto';
+import { ResultDto } from './dto/response/result.dto';
+import { IUserJWT } from '../../common/interface/user-jwt.interface';
+import { doesUserHaveAuthGroup } from '../../common/helper/auth.helper';
+
+@Injectable()
+export class ShoppingCartService {
+  private readonly logger = new Logger(ShoppingCartService.name);
+  constructor(
+    @InjectRepository(Application)
+    private readonly applicationRepository: Repository<Application>,
+  ) {}
+
+  /**
+   * Adds selected applications to the shopping cart by updating their status to `IN_CART`.
+   * The method handles permission checks based on the current user's role and updates the status accordingly.
+   *
+   * @param currentUser - The current user's JWT payload containing user identification and roles.
+   * @param dto - An object containing the application IDs to add to the cart and the associated company ID.
+   * @returns A promise that resolves with a `ResultDto` containing arrays of `success` with IDs of applications successfully added to the cart, and `failure` with IDs of applications that failed to be added.
+   */
+
+  @LogAsyncMethodExecution()
+  async addToCart(
+    currentUser: IUserJWT,
+    { applicationIds, companyId }: AddToShoppingCartDto & { companyId: number },
+  ): Promise<ResultDto> {
+    const { orbcUserAuthGroup } = currentUser;
+    if (orbcUserAuthGroup === ClientUserAuthGroup.COMPANY_ADMINISTRATOR) {
+      return await this.updateApplicationStatus(
+        { applicationIds, companyId },
```

Comment:
What's the meaning behind not adding userGUID here while adding userGUID for permit applicant?

Reply:
Permit Applicant only gets to see their own applications in the cart. Company Admins can see everybody's applications that's in cart. i.e., Company admins have a collective company cart.

Reply:
ok I see, got it